src/Entity/ItemId.php

```diff
@@ -103,4 +103,21 @@ public static function newFromNumber( $numericId ) {
 		return new self( 'Q' . $numericId );
 	}
 
+	/**
+	 * CAUTION: Use the full string serialization whenever you can and avoid using numeric IDs.
+	 *
+	 * @param string $repositoryName
+	 * @param int|float|string $numericId
+	 *
+	 * @return self
+	 * @throws InvalidArgumentException
+	 */
+	public static function newForeignFromNumber( $repositoryName, $numericId ) {
```

Comment:
this is not just "...FromNumber" (repository name + a number)  though can't think of a better name right now 

with "...FromNumber", I somewhat would think the $numericId would be the first parameter (the most important thing)